float32[] left_touch
float32[] right_touch
float32 angle_touch